source/kraken/apply_disruption.cpp

```diff
@@ -115,6 +115,21 @@ struct apply_impacts_visitor : public boost::static_visitor<> {
     }
 };
 
+static type::ValidityPattern compute_vp(const std::vector<boost::posix_time::time_period>& periods,
+        const boost::gregorian::date_period& production_period) {
+    type::ValidityPattern vp{production_period.begin()}; // bitset are all initialised to 0
+    for (const auto& period: periods){
+        // we may impact vj's passed midnight
+        bg::day_iterator titr(period.begin().date());
+        for (; titr <= period.end().date(); ++titr) {
```

Comment:
This loop will cause problem with pass midnight vj. For the moment, as we only support one day update, we can just set the first day. But we'll need to find another approach to handle stop point cancelation in chaos.